### PR TITLE
DAOS-3752 test: Adding test to verify aggregation on server restart

### PR DESCRIPTION
**Pull request by saurabhtandan**
Adding test to verify aggregation after io is performed for
server restart.
Adding two files as part of the changes to this PR
 - io_aggregation.py
 - io_aggregation.yaml
Modifying ior_testbase.py
Adding festure to create and destroy snapshot in
daos_utils.py
Adding feature to parse through the log file in
general_utils.py

Test-tag-hw-small: pr,hw,small ioaggregation

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/ior_test_base.py [hunk 1]
```diff
@@ -166,16 +167,17 @@ def run_ior_with_pool(self, intercept=None, test_file_suffix="",
             self.dfuse = None
         return out
 
-    def update_ior_cmd_with_pool(self):
+    def update_ior_cmd_with_pool(self, create_cont=True):
         """Update ior_cmd with pool."""
         # Create a pool if one does not already exist
         if self.pool is None:
             self.create_pool()
         # Always create a container
```
Comment (shimizukko) [inline src/tests/ftest/util/ior_test_base.py:175]: Now we don't always create a container, so we don't need this comment.

File: src/tests/ftest/util/test_utils_container.py [hunk 2]
```diff
@@ -368,6 +369,103 @@ def create(self, uuid=None, con_in=None, acl_file=None):
         self.uuid = self.container.get_uuid_str()
         self.log.info("  Container created with uuid %s", self.uuid)
 
+    @fail_on(DaosApiError)
+    def create_snap(self, pool=None, cont=None, snap_name=None, epoch=None,
```
Comment (shimizukko) [inline src/tests/ftest/util/test_utils_container.py:373]: I don't think we should provide pool, cont, svc, and sys_name as parameters. After TestContainer object is created, we expect the caller to call create, which sets the container UUID to its member. We also expect the pool to exist before creating a TestContainer object, so pool, svc, and sys_name of its pool member should be used here. So I think line 386 to 393 should be removed and the values in kwargs should be adjusted to use the members.

File: src/tests/ftest/util/test_utils_container.py [hunk 3]
```diff
@@ -368,6 +369,103 @@ def create(self, uuid=None, con_in=None, acl_file=None):
...
+
+        else:
+            self.log.error(
+                "Error: Undefined control_method: %s",
+                self.control_method.value)
+
+        self.epoch = output.split()[1]
+
+    @fail_on(DaosApiError)
+    def destroy_snap(self, pool=None, cont=None, snap_name=None, epoch=None,
```
Comment (shimizukko) [inline src/tests/ftest/util/test_utils_container.py:418]: Same comment as above. It doesn't make sense to provide this method the ability to do destroy snapshot work with whatever pool or container the caller specifies. That feature is already provided by container_destroy_snap method. The methods in this class should deal with the particular TestContainer object.

File: src/tests/ftest/util/test_utils_container.py [hunk 4]
```diff
@@ -368,6 +369,103 @@ def create(self, uuid=None, con_in=None, acl_file=None):
...
+            sys_name = self.pool.name.value
+        if self.control_method.value == self.USE_DAOS and self.daos:
+            # create snapshot using daos utility
+            kwargs = {
+                "pool" : pool.uuid,
+                "cont" : cont,
+                "snap_name" : snap_name,
+                "epoch" : epoch,
+                "svc" : svc,
+                "sys_name": sys_name,
```
Comment (shimizukko) [inline src/tests/ftest/util/test_utils_container.py:402]:
```suggestion
                "pool" : self.pool.uuid,
                "cont" : self.uuid,
                "snap_name" : snap_name,
                "epoch" : epoch,
                "svc" : ",".join(str(rank) for rank in self.pool.svc_ranks),
                "sys_name": self.pool.name.value,
```

File: src/tests/ftest/util/ior_test_base.py (hunk 1 above)
Comment (saurabhtandan) [reply src/tests/ftest/util/ior_test_base.py]: Done

File: src/tests/ftest/util/test_utils_container.py (hunk 2 above)
Comment (saurabhtandan) [reply src/tests/ftest/util/test_utils_container.py]: Done

File: src/tests/ftest/util/test_utils_container.py (hunk 3 above)
Comment (saurabhtandan) [reply src/tests/ftest/util/test_utils_container.py]: Done

File: src/tests/ftest/util/test_utils_container.py (hunk 4 above)
Comment (saurabhtandan) [reply src/tests/ftest/util/test_utils_container.py]: Done

File: src/tests/ftest/io/io_aggregation.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up test before executing"""
+        super(IoAggregation, self).setUp()
+        self.dmg = self.get_dmg_command()
+        self.daos_cmd = DaosCommand(self.bin)
+
+    def display_free_space(self):
```
[the remark this answers is not in the document]
Comment (saurabhtandan) [reply src/tests/ftest/io/io_aggregation.py]: Done

File: src/tests/ftest/util/test_utils_container.py [hunk 6]
```diff
@@ -383,23 +382,23 @@ def create_snap(self, pool=None, cont=None, snap_name=None, epoch=None,
...
-        if not sys_name:
-            sys_name = self.pool.name.value
+#        if not pool:
+#            pool = self.pool
+#        if not cont:
+#            cont = self.uuid
+#        if not svc:
+#            svc = ",".join(str(rank) for rank in self.pool.svc_ranks)
+#        if not sys_name:
+#            sys_name = self.pool.name.value
```
Comment (shimizukko) [inline src/tests/ftest/util/test_utils_container.py:392]: Is there any reason for keeping these lines?

File: src/tests/ftest/util/test_utils_container.py [hunk 7]
```diff
@@ -431,24 +429,24 @@ def destroy_snap(self, pool=None, cont=None, snap_name=None, epoch=None,
...
```
Comment (shimizukko) [inline src/tests/ftest/util/test_utils_container.py:439]: Is there any reason for keeping these lines?

File: src/tests/ftest/util/test_utils_container.py (hunk 6 above)
Comment (saurabhtandan) [reply src/tests/ftest/util/test_utils_container.py]: Ah got pushed accidentally with commented lines. Thanks for catching it. Should be resolved now.

**Pull request merged**